### PR TITLE
Fix/pom.xmls

### PR DESCRIPTION
**Pull request by koppor**
This streamlines the pom.xml files and also includes maven-javadoc-plugin whereever possible.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -169,6 +169,206 @@
...
+                    <version>0.12</version>
+                    <configuration>
+                        <message>Update winery artifacts (${project.version})</message> <!-- git commit message -->
+                        <noJekyll>true</noJekyll> <!-- disable webpage processing -->
+                        <force>true</force>
+                        <merge>true</merge>
+                        <outputDirectory>${project.basedir}/target/mvn-repo</outputDirectory> <!-- matches distribution management repository url above -->
+                        <branch>refs/heads/master</branch> <!-- remote branch name -->
+                        <includes><include>**/*</include></includes>
+                        <repositoryOwner>OpenTOSCA</repositoryOwner>
```
Comment (lharzenetter) [inline pom.xml]: Is this intended to be included into eclipse?

File: pom.xml [hunk 2]
```diff
@@ -169,6 +169,181 @@
...
+                    <artifactId>p2-maven-plugin</artifactId>
+                    <version>1.3.0</version>
+                    <executions>
+                        <execution>
+                            <id>default-cli</id>
+                            <configuration>
+                                <featureDefinitions>
+                                    <feature>
+                                        <id>winery</id>
+                                        <version>2.0.0</version>
```
Comment (lharzenetter) [inline pom.xml]: Shouldn't this be `2.0.0-SNAPSHOT`?

Comment (lharzenetter) [reply pom.xml]: @miwurster what is your opinion?

Comment (miwurster) [reply pom.xml]: No, it should not. This configuration relates to Eclipse update site we are able to build. An Eclipse update site works totally different than Maven, e.g., no awareness of SNAPSHOT.

**Pull request merged**